### PR TITLE
5585 chattiin nakyviin oma ohjaaja

### LINKED ISSUE #5585: Chattiin näkyviin oma ohjaaja
Oma ohjaaja saataville chattiin

- Henkilöille ja huoneille omat tabit chattiin
- Oma ohjaaja ko. tabiin
- Avataan chat-ikkuna henkilön klikkauksesta
- Chat-ikkunan avaus myös kursseille ja opinnot-näkymään

### PR DESCRIPTION
**Pull request by avukkonen**
Closes #5585

### CODE REVIEW

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/chat/chat.tsx [hunk 1]
```diff
@@ -209,6 +237,123 @@ class Chat extends React.Component<IChatProps, IChatState> {
...
+              }),
+              "callback"
+            )().then((result: SummaryStudentsGuidanceCouncelorsType[]) => {
+              result.forEach((studentsGuidanceCouncelor) => {
+                if (
+                  !studentsGuidanceCouncelors.some(
+                    (existingStudentCouncelor) =>
+                      existingStudentCouncelor.userEntityId ==
+                      studentsGuidanceCouncelor.userEntityId
+                  )
```
Comment (TeroOtavia) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/chat/chat.tsx:274]: Why not using === operator?

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/chat/chat.tsx [hunk 2]
```diff
@@ -209,6 +237,123 @@ class Chat extends React.Component<IChatProps, IChatState> {
...
+                ) {
+                  studentsGuidanceCouncelors.push(studentsGuidanceCouncelor);
+                  studentsGuidanceCouncelors.sort((x, y) => {
+                    const a = x.lastName.toUpperCase(),
+                      b = y.lastName.toUpperCase();
+                    return a == b ? 0 : a > b ? 1 : -1;
+                  });
+                }
```
Comment (TeroOtavia) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/chat/chat.tsx:282]: You can sorting after forEach is done and before setState. Every loop is unnecessary.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/chat/chat.tsx [hunk 3]
```diff
@@ -209,6 +237,123 @@ class Chat extends React.Component<IChatProps, IChatState> {
...
+        jids.map(async (contact: IChatContact) => {
+          await promisify(
+            mApi().chat.userInfo.read(contact.jid.split("@")[0], {}),
+            "callback"
+          )().then((user: IChatContact) => {
+            chatRoster.push({
+              ...{ jid: contact.jid },
+              ...user,
+            });
+          });
```
Comment (TeroOtavia) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/chat/chat.tsx:337]: Here you are adding new item to array. To clear this a bit, spread object first and then change jid to what you want it to really be.
```ts
chatRoster.push({...user, jid: contact.jid});

```

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/chat/chat.tsx [hunk 4]
```diff
@@ -607,10 +770,9 @@ class Chat extends React.Component<IChatProps, IChatState> {
   }
 
   /**
-   * @param status
-   * @param condition
+   * @param status strophe status
    */
-  onConnectionStatusChanged(status: Strophe.Status, condition: string) {
+  onConnectionStatusChanged(status: Strophe.Status) {
```
Comment (TeroOtavia) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/chat/chat.tsx:775]: Missing function documentation.

**Review by TeroOtavia — CHANGES_REQUESTED** [review]

**Review by TeroOtavia — APPROVED** [review]

**Review by jjlankinen — APPROVED** [review]

**Pull request merged**